packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/FLTGoogleMapHeatmapController.m

```diff
@@ -0,0 +1,186 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "FLTGoogleMapHeatmapController.h"
+#import "FLTGoogleMapJSONConversions.h"
+@import GoogleMapsUtils;
+
+@interface FLTGoogleMapHeatmapController ()
+
+// The heatmap tile layer this controller handles.
+@property(nonatomic, strong) GMUHeatmapTileLayer *heatmapTileLayer;
+
+// The map view owned by GoogmeMapController.
+@property(nonatomic, weak) GMSMapView *mapView;
+
+@end
+
+// Static wrapper for interpreting heatmap options.
+static void InterpretHeatmapOptions(FLTGoogleMapHeatmapController *self,
+                                    NSDictionary<NSString *, id> *options) {
+  [self interpretHeatmapOptions:options];
```

Comment:
I think you misunderstood my earlier comment about having a static method. The point isn't to have a static method just for the sake of doing so, it's that Objective-C code should not call methods on `self` in the context of `init*` or `dealloc` methods, at all. Hiding that in a static wrapper doesn't help, it just makes it slightly less obvious that the code is dangerous.

This method should not be doing any method invocation on `self`. It should be doing whatever extraction logic is necessary, then directly setting instance variable values.

https://google.github.io/styleguide/objcguide.html#avoid-messaging-the-current-object-within-initializers-and--dealloc